File: android/src/main/java/com/guichaguri/trackplayer/service/models/NowPlayingMetadata.java
package com.guichaguri.trackplayer.service.models;

import android.content.Context;
import android.os.Bundle;

public class NowPlayingMetadata extends TrackMetadata {

    public double elapsedTime;

    public NowPlayingMetadata(Context context, Bundle bundle, int ratingType) {
        setMetadata(context, bundle, ratingType);
    }

    @Override
    public void setMetadata(Context context, Bundle bundle, int ratingType) {
        super.setMetadata(context, bundle, ratingType);

        elapsedTime = bundle.getDouble("elapsedTime", 0);
    }
    
}

File: android/src/main/java/com/doublesymmetry/trackplayer/model/NowPlayingMetadata.kt
package com.doublesymmetry.trackplayer.model

import android.content.Context
import android.os.Bundle

class NowPlayingMetadata(context: Context, bundle: Bundle?, ratingType: Int) : TrackMetadata() {
    var elapsedTime = 0.0
    override fun setMetadata(context: Context, bundle: Bundle?, ratingType: Int) {
        super.setMetadata(context, bundle, ratingType)
        elapsedTime = bundle!!.getDouble("elapsedTime", 0.0)
    }

    init {
        setMetadata(context, bundle, ratingType)
    }
}